controllers/user.controller.js

```diff
@@ -0,0 +1,247 @@
+            message: "Empty data!"
+        });
+    }
+
+    const userID = req.body.userID;
+    try {
+        user.query('SELECT token FROM mailPass where userID = ?', userID,
+            function (err, result) {
+                if (err) {
+                    throw err;
+                } else if (!result[0]) {
+                    return res.status(505).json({ message: "Database problem" });
+                } else if (result[0].token) {
+                    const token = result[0].token;
+                    if (token == req.body.token) {
+                        user.query(`UPDATE mailPass SET token = NULL WHERE userID = ?`,
+                            userID, (error, results) => {
+                                if (error) {
+                                    throw error;
+                                } else {
+                                    return res.status(200).json({ message: "By!!!" });
+                                }
+                        });
+                    }
+                } 
+        });
+    } catch (err) {
+        return res.status(400).json({ message: err.message });
+    }
+}
+
+let getUserInfo = async function (req, res) {
+    const userID = req.query.userID;
+    const token = req.query.token;
+    if (userID && token) {
+        user.query(`SELECT * FROM mailPass WHERE userID = ? AND token = ?`,
+            [userID, token],
+            function (err, results) {
+                if (err) {
+                    throw err;
+                } else if (!results[0]) {
+                    return res.status(400).json({
+                        message: "incorrect token"
+                    });
+                } else if (results[0].mail) {
+                    user.query('SELECT * FROM users WHERE userID = ?',
+                        userID,
+                        function (err, result) {
+                            if (err) {
+                                throw err;
+                            } else if (!result[0]) {
+                                return res.status(400).json({
+                                    message: "Incorrect token"
+                                });
+                            } else {
+                                return res.status(200).json({
+                                    name: result[0].name,
+                                    surname: result[0].surname,
+                                    age: result[0].age,
+                                    mail: results[0].mail
+                                });
+                            }
+                        }
+                );
+            }
+        });
+    } else {
+        res.status(401).json({
+            message: "Empty data"
+        });
+    }
+}
+
+let getUserTables = async function (req, res) {
+    const userID = req.query.userID;
+    const token = req.query.token;
+    if (userID && token) {
+        user.query(`SELECT * FROM mailPass WHERE userID = ? AND token = ?`,
```

Comment:
things like this should be implemented as route middleware.
For more info please see following:
https://expressjs.com/en/guide/using-middleware.html